### PR TITLE
[2/3] Applications endpoint

### PR DESCRIPTION
**Pull request by zsinx6**
This PR is a continuation of the https://github.com/omnivector-solutions/jobbergate-api-fastapi/pull/3
It covers for the DELETE and GET in the /applications endpoint.

The next will cover the PUT.

### CODE REVIEW

File: jobbergateapi2/apps/applications/routers.py [hunk 1]
```diff
@@ -65,5 +67,84 @@ async def applications_create(
...
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Application {application_id=} not found for the user with id={current_user.id}",
+        )
+    application = Application.parse_obj(raw_application)
+    delete_query = applications_table.delete().where(where_stmt)
+    await database.execute(delete_query)
+    application_location = (
+        f"{settings.S3_BASE_PATH}/TEST/applications/{application.id}/jobbergate.tar.gz"
+        # f"{S3_BASE_PATH}/{application.owner_id}/applications/{application.id}/jobbergate.tar.gz"
```
Comment (zsinx6) [inline jobbergateapi2/apps/applications/routers.py:96]: This is here for testing only, and the line below will be the one used when released

File: jobbergateapi2/apps/applications/routers.py [hunk 2]
```diff
@@ -65,5 +67,84 @@ async def applications_create(
...
+    s3_client = boto3.client("s3")
+    where_stmt = (applications_table.c.id == application_id) & (
+        applications_table.c.application_owner_id == current_user.id
+    )
+    get_query = applications_table.select().where(where_stmt)
+    raw_application = await database.fetch_one(get_query)
+    if not raw_application:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Application {application_id=} not found for the user with id={current_user.id}",
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/applications/routers.py:89]: remove `=`

Comment (zsinx6) [reply jobbergateapi2/apps/applications/routers.py:89]: Why? When we use `variable=` inside `{}` of fstrings it expands to `variable=variable_value`

File: jobbergateapi2/apps/applications/routers.py [hunk 3]
```diff
@@ -65,5 +67,84 @@ async def applications_create(
...
+    """
+    query = applications_table.select().where(
+        (applications_table.c.application_owner_id == current_user.id)
+        & (applications_table.c.id == application_id)
+    )
+    raw_application = await database.fetch_one(query)
+    if not raw_application:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Application {application_id=} not found for the user with id={current_user.id}",
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/applications/routers.py:142]: I think this is another typo ... possibly some syntax I'm unaware of .... lol

File: jobbergateapi2/apps/applications/routers.py (hunk 2 above)
Comment (jamesbeedy) [reply jobbergateapi2/apps/applications/routers.py:89]: 🤦  - nice

File: jobbergateapi2/apps/applications/routers.py [hunk 4]
```diff
@@ -65,5 +67,84 @@ async def applications_create(
...
+
+
+@router.get(
+    "/applications/",
+    description="Endpoint to list applications",
+    response_model=List[Application],
+)
+async def applications_get(all: Optional[bool] = Query(None), current_user: User = Depends(get_current_user)):
+    """
+    Endpoint to list all the applications from the authenticated user
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/applications/routers.py:111]: change single line doc strings to use imperative mood

**Comment by jamesbeedy** [comment]
👍

**Pull request merged**